src/cargo/ops/cargo_package.rs

```diff
@@ -82,48 +83,33 @@ struct GitVcsInfo {
     sha1: String,
 }
 
+/// Packages a single package in a workspace, returning the resulting tar file.
+///
+/// Ignores `opts.list`.
 pub fn package_one(
     ws: &Workspace<'_>,
     pkg: &Package,
     opts: &PackageOpts<'_>,
-) -> CargoResult<Option<FileLock>> {
-    let gctx = ws.gctx();
-    let mut src = PathSource::new(pkg.root(), pkg.package_id().source_id(), gctx);
-    src.update()?;
+) -> CargoResult<FileLock> {
+    let ar_files = build_ar_list(ws, pkg, opts)?;
+    let tarball = create_package(ws, pkg, ar_files)?;
```

Comment:
If you want to reduce the risk of conflicts, I think this commit stands enough on its own that it can be split out and not wait for the rest of the review